test/lib.bash

```diff
@@ -53,23 +53,29 @@ function make_temporary_gopath {
   cp -rv "$(go env GOPATH)/bin" "${tmp_gopath}"
   logger.info "Temporary GOPATH is: ${tmp_gopath}"
   export GOPATH="$tmp_gopath"
+  export PATH="$GOPATH/bin":$PATH
 }
 
-function run_knative_serving_tests {
-  (
+function remove_temporary_gopath {
+  if [[ "$GOPATH" =~ .*gopath-[0-9a-zA-Z]{10} ]]; then
+    logger.info "Removing GOPATH: ${GOPATH}"
+    rm -rf "${GOPATH}"
+  fi
+}
+
+function checkout_knative_serving {
   local knative_version=$1
+  # Setup a temporary GOPATH to safely check out the repository without breaking other things.
   make_temporary_gopath
 
-  # Save the rootdir before changing dir
-  rootdir="$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")"
-
   # Checkout the relevant code to run
-  mkdir -p "$GOPATH/src/knative.dev"
-  git clone --branch "release-${knative_version}" \
-    --depth=1 https://github.com/openshift/knative-serving.git \
-    "${GOPATH}/src/knative.dev/serving"
-  pushd "${GOPATH}/src/knative.dev/serving" || return $?
+  export KNATIVE_SERVING_HOME="$GOPATH/src/knative.dev/serving"
+  mkdir -p "$KNATIVE_SERVING_HOME"
+  git clone -b "release-${knative_version}" --depth 1 https://github.com/openshift/knative-serving.git "$KNATIVE_SERVING_HOME"
+  git describe --always --tags --dirty
```

Comment:
This is the single minor issue that I have with this code. Invoking it just after checkout will always result in clean checkout, `--dirty` flag will report clean, always. Then we modify repository in some way (the `rm` command). 

Better will be to put this directly before test execution.

Reply:
Other than this it looks okay to me. This is minor.